lib/benchee/conversion/memory.ex

```diff
@@ -0,0 +1,197 @@
+                      long:      "Megabytes"
+                   },
+      kilobyte:      %Unit{
+                      name:      :kilobyte,
+                      magnitude: @bytes_per_kilobyte,
+                      label:     "KB",
+                      long:      "Kilobytes"
+                   },
+      byte: %Unit{
+                      name:      :byte,
+                      magnitude: 1,
+                      label:     "B",
+                      long:      "Bytes"
+                   }
+  }
+
+  @doc """
+  Scales a memory value in bytes into a larger unit if appropriate
+
+  ## Examples
+
+    iex> {value, unit} = Benchee.Conversion.Memory.scale(1)
+    iex> value
+    1.0
+    iex> unit.name
+    :byte
+
+    iex> {value, unit} = Benchee.Conversion.Memory.scale(1_234)
+    iex> value
+    1.205078125
+    iex> unit.name
+    :kilobyte
+
+    iex> {value, unit} = Benchee.Conversion.Memory.scale(11_234_567_890.123)
+    iex> value
+    10.463006692121736
+    iex> unit.name
+    :gigabyte
+
+    iex> {value, unit} = Benchee.Conversion.Memory.scale(1_111_234_567_890.123)
+    iex> value
+    1.0106619519229962
+    iex> unit.name
+    :terabyte
+  """
+  def scale(memory) when memory >= @bytes_per_terabyte do
+    scale_with_unit memory, :terabyte
+  end
+  def scale(memory) when memory >= @bytes_per_gigabyte do
+    scale_with_unit memory, :gigabyte
+  end
+  def scale(memory) when memory >= @bytes_per_megabyte do
+    scale_with_unit memory, :megabyte
+  end
+  def scale(memory) when memory >= @bytes_per_kilobyte do
+    scale_with_unit memory, :kilobyte
+  end
+  def scale(memory) do
+    scale_with_unit memory, :byte
+  end
+
+  # Helper function for returning a tuple of {value, unit}
+  defp scale_with_unit(memory, unit) do
+    {scale(memory, unit), unit_for(unit)}
+  end
+
+  @doc """
+  Get a unit by its atom representation.
+
+  ## Examples
+
+      iex> Benchee.Conversion.Memory.unit_for :gigabyte
+      %Benchee.Conversion.Unit{
+          name:      :gigabyte,
+          magnitude: 1_073_741_824,
+          label:     "GB",
+          long:      "Gigabytes"
+      }
+  """
+  def unit_for(unit) do
+    Scale.unit_for @units, unit
+  end
+
+  @doc """
+  Scales a memory value in bytes into a value in the specified unit
+
+  ## Examples
+
+      iex> Benchee.Conversion.Memory.scale(12345, :kilobyte)
+      12.0556640625
+
+      iex> Benchee.Conversion.Memory.scale(12345, :megabyte)
+      0.011773109436035156
+
+      iex> Benchee.Conversion.Memory.scale(123_456_789, :gigabyte)
+      0.11497809458523989
+
+  """
+  def scale(count, unit) do
+    Scale.scale count, unit, __MODULE__
+  end
+
+  @doc """
+  Finds the best unit for a list of memory units. By default, chooses the most common
+  unit. In case of tie, chooses the largest of the most common units.
+
+  Pass `[strategy: :smallest]` to always return the smallest unit in the list.
+  Pass `[strategy: :largest]` to always return the largest unit in the list.
```

Comment:
I think we support 4 strategies, weird to just mention 2. But that might be duplicated from other places anyhow...

Reply:
good point